std/algorithm/iteration.d

```diff
@@ -2864,6 +2864,9 @@ Two adjacent separators are considered to surround an empty element in
 the split range. Use $(D filter!(a => !a.empty)) on the result to compress
 empty elements.
 
+The predicate is passed to $(XREF functional,binaryFun), and can accept a string,
+function pointer, lambda, or function literal.
+
```

Comment:
There are many things that work here, including delegates, functors, functions, templates, etc.

I think it's probably best to identify that strings here are a special case. Everything else is an alias to a 
"callable" entity. Perhaps: "The predicate can be either a string literal lambda (see $(XREF functional,binaryFun) for more details) or an alias that can be called with $(D pred(element, s))"